chainerx_cc/chainerx/cuda/cuda_device/linalg.cu

```diff
@@ -204,5 +245,60 @@ public:
 
 CHAINERX_CUDA_REGISTER_KERNEL(InverseKernel, CudaInverseKernel);
 
+class CudaCholeskyKernel : public CholeskyKernel {
+public:
+    void Call(const Array& a, const Array& out) override {
+        Device& device = a.device();
+        device.CheckDevicesCompatible(a, out);
+        Dtype dtype = a.dtype();
+        CudaSetDeviceScope scope{device.index()};
+
+        CHAINERX_ASSERT(a.ndim() == 2);
+        CHAINERX_ASSERT(out.ndim() == 2);
+        CHAINERX_ASSERT(a.shape()[0] == a.shape()[1]);
+
+        // potrf (cholesky) stores result in-place, therefore copy ``a`` to ``out`` and then pass ``out`` to the routine
+        device.backend().CallKernel<CopyKernel>(Tril(a, 0), out);
+
+        Array out_contiguous = AsContiguous(out);
```

Comment:
If `out` is not contiguous, a copy is created here, meaning that nothing is written to the argument `out`. We need a temporary out array for the computations and then write the results back to `out`, in case `out` is not contiguous.

Reply:
Added assert to kernels to ensure that passed  `out` array is contiguous.